bsms.mediawiki

```diff
@@ -63,39 +63,39 @@ The Signer is a participating member in the multisig. Its responsibilities inclu
 * The first line in the record must be the <tt>TOKEN</tt>. The second line must be the <tt>KEY</tt>. The <tt>KEY</tt> is an XPUB plus its key origin information, written in the descriptor-defined format, i.e.: <tt>[{master key fingerprint}/{derivation path}]{XPUB}</tt>. The third line is a text description of the key, 80 characters maximum. The fourth line must be a <tt>SIG</tt>, whereas <tt>SIG</tt> is the signature generated by using the private key associated with the XPUB to sign the first three lines. The signature should follow [https://github.com/bitcoin/bips/blob/master/bip-0322.mediawiki BIP-0322], legacy format accepted.
-* Finally, the Signer encrypts the key record with <tt>ENCRYPTION_KEY</tt>. The Signer calculates the 256-bit Initialization Vector (<tt>IV</tt>) for the XPUB, and appends it to the end of the encrypted text.
+* The Signer calculates the Message Authentication Code (<tt>MAC</tt>) for the record. The <tt>MAC</tt> doubles as the Initialization Vector (<tt>IV</tt>) for the encryption.
+* Finally, the Signer encrypts the key record with <tt>ENCRYPTION_KEY</tt> and <tt>IV</tt>.  The Signer appends the MAC to the end of the encrypted text.
 
 ====Round 2====
 
 =====Coordinator=====
 
 * The Coordinator gathers key records from all participating Signers. Abort the setup if there are fewer than <tt>N</tt> unique key records when the wallet setup session expires.
-* For each key record, the Coordinator extracts the <tt>IV</tt> from the data, then decrypts the ciphertext using <tt>ENCRYPTION_KEY</tt> and <tt>IV</tt>.
-* The Coordinator verifies that the included <tt>IV</tt> is valid given the decrypted data.
+* For each key record, the Coordinator extracts the <tt>MAC</tt> from the data, sets <tt>IV</tt> to <tt>MAC</tt>, then decrypts the ciphertext using <tt>ENCRYPTION_KEY</tt> and <tt>IV</tt>.
+* The Coordinator verifies that the included <tt>MAC</tt> is valid given the plaintext.
 * The Coordinator verifies that the included <tt>SIG</tt> is valid given the <tt>KEY</tt>.
 * If all key records look good, the Coordinator fills in all necessary information to generate a descriptor record, which is simply the descriptor string plus a <tt>CHECKSUM</tt>, all in one line. The <tt>CHECKSUM</tt> has [https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md#checksums BECH32 encoding].
-* The Coordinator encrypts the descriptor record with <tt>ENCRYPTION_KEY</tt>. The Coordinator calculates the 256-bit <tt>IV</tt> for the descriptor, and appends it to the end of the encrypted text.
+* The Coordinator calculates the <tt>MAC</tt> for the record. The <tt>MAC</tt> doubles as the <tt>IV</tt> for the encryption. 
+* The Coordinator encrypts the descriptor record with <tt>ENCRYPTION_KEY</tt> and <tt>IV</tt>. The Coordinator appends the MAC to the end of the encrypted text.
```

Comment:
`The Coordinator encrypts the descriptor record with the <tt>ENCRYPTION_KEY</tt>...`